resultshandling/printer/prometheusprinter.go

```diff
@@ -25,74 +29,41 @@ func (prometheusPrinter *PrometheusPrinter) Score(score float32) {
-			}
-			apiVersion,ok := k8sObj[`apiVersion`].(string)
-			if (!ok) {
-				return errors.New("Found object with non string apiVersion")
-			}
-			gvk := fmt.Sprintf("%s/%s",apiVersion,kind)
-			metadata,ok := k8sObj[`metadata`].(map[string]interface{})
-			if (!ok) {
-				return errors.New("Found object with non convertable metadata")
-			}
-			name,ok := metadata[`name`].(string)
-			if (!ok) {
-				return errors.New("Found metadata with non string name")
-			}
-			namespace,ok := metadata[`namespace`].(string)
-			if (!ok) {
-				namespace = ""
-			}
-			if (objs[gvk] == nil) {
-				objs[gvk] = make(map[string]map[string]int)
-			}
-			if (objs[gvk][namespace] == nil) {
-				objs[gvk][namespace] = make(map[string]int)
-			}
-			objs[gvk][namespace][name]++
-		}
+func (printer *PrometheusPrinter) printResources(allResources map[string]workloadinterface.IMetadata, resourcesIDs *reporthandling.ResourcesIDs, frameworkName, controlName string) {
+	printer.printDetails(allResources, resourcesIDs.GetFailedResources(), frameworkName, controlName, "failed")
+	printer.printDetails(allResources, resourcesIDs.GetWarningResources(), frameworkName, controlName, "excluded")
+	if printer.verboseMode {
+		printer.printDetails(allResources, resourcesIDs.GetPassedResources(), frameworkName, controlName, "passed")
 	}
-	for gvk, namespaces := range objs {
-		for namespace, names := range namespaces {
-			for name, value := range names {
-				fmt.Fprintf(printer.writer, "# Failed object from %s control %s\n", frameworkName, controlName)
-				if namespace != "" {
-					fmt.Fprintf(printer.writer, "kubescape_object_failed_count{framework=\"%s\",control=\"%s\",namespace=\"%s\",name=\"%s\",groupVersionKind=\"%s\"} %d\n", frameworkName, controlName, namespace, name, gvk, value)
-				} else {
-					fmt.Fprintf(printer.writer, "kubescape_object_failed_count{framework=\"%s\",control=\"%s\",name=\"%s\",groupVersionKind=\"%s\"} %d\n", frameworkName, controlName, name, gvk, value)
-				}
-			}
+
+}
+func (printer *PrometheusPrinter) printDetails(allResources map[string]workloadinterface.IMetadata, resourcesIDs []string, frameworkName, controlName, status string) {
+	fmt.Fprintf(printer.writer, "# Number of resources found as part of %s control \"%s\"\nkubescape_resources_found_count{framework=\"%s\",control=\"%s\"} %d\n", status, controlName, frameworkName, controlName, len(resourcesIDs))
+	for _, resourceID := range resourcesIDs {
+		resource := allResources[resourceID]
+		fmt.Fprintf(printer.writer, "# %s object from \"%s\" control \"%s\"\n", strings.Title(status), frameworkName, controlName)
+		if resource.GetNamespace() != "" {
+			fmt.Fprintf(printer.writer, "kubescape_object_%s_count{framework=\"%s\",control=\"%s\",namespace=\"%s\",name=\"%s\",groupVersionKind=\"%s\"} %d\n", status, frameworkName, controlName, resource.GetNamespace(), resource.GetName(), fmt.Sprintf("%s/%s", resource.GetApiVersion(), resource.GetKind()), 1)
```

Comment:
The hardcoded '1' here is wrong. There are multiple output lines with identical contents in the example output given. e.g. 
```
kubescape_object_passed_count{framework="MITRE",control="Cluster-admin binding",namespace="kubesystem",name="storage-provisioner",groupVersionKind="/ServiceAccount"} 1
```
and for this prometheus will drop the fact there are two of these and treat it as though the result was 1 when it should be 2. e.g.
```
kubescape_object_passed_count{framework="MITRE",control="Cluster-admin binding",namespace="kubesystem",name="storage-provisioner",groupVersionKind="/ServiceAccount"} 2
```

Reply:
I guess it's because we are looking at the GroupVersion and kind without considering other resources that cause the control to fail.
In this example, the same `ServiceAccount` failed because it was associated to more than one `RoleBinding`/`Role`. If we 
will refer to the ID instead of just the GroupVersion/kind, each resource will be unique, and then the counter can be hardcoded 1.

It will look as following:
```
resource := allResources[resourceID]
		fmt.Fprintf(printer.writer, "# %s object from \"%s\" control \"%s\"\n", strings.Title(status), frameworkName, controlName)
		if resource.GetNamespace() != "" {
			fmt.Fprintf(printer.writer, "kubescape_object_%s_count{framework=\"%s\",control=\"%s\",namespace=\"%s\",name=\"%s\",groupVersionKind=\"%s\"} %d\n", status, frameworkName, controlName, resource.GetNamespace(), resource.GetName(), resource.GetID(), 1)
```

With kubernetes objects the returned ID will look as [following](https://github.com/armosec/k8s-interface/blob/master/workloadinterface/workloadmethods.go#L347)

With more complicated controls (like a serviceAccount that fails because of the Roles) the ID will [look as following](https://github.com/armosec/k8s-interface/blob/master/workloadinterface/regoresponsevectorobject.go#L141)

What do you think?